### DISCUSSION TITLE
What is the behavior when the certificate expires during an established TLS connection?

### DISCUSSION THREAD

**Discussion by duxiaocheng** [openssl/openssl#23196] [Q&A] [answered]
What is the behavior for openssl when the certificate expires during an established TLS connection?
Is there any document about this scenario?
Thanks!

**Comment by mattcaswell** [comment, marked as answer]
> What is the behavior for openssl when the certificate expires during an established TLS connection?

Certificates are verified during the handshake and then never looked at again unless a renegotiation occurs. The certificate only has to be valid at the time of the verification. If it subsequently expires and the connection is still active then the connection remains active.

**Reply by duxiaocheng** [reply]
Thank you very much!
